### ISSUE TITLE
Some Litedown is parsed inside of BBCode code blocks

### ISSUE THREAD

**Issue by johnd0e** [s9e/TextFormatter#169]
I have some bbcode tags for syntax highlighting: [code],[codebox].
Problem is that Litedown break thinhs trying to process their content.
How can i prevent that?

**Comment by JoshyPHP** [comment]
It depends on the context, in general it works fine without any configuration. What's the BBCode definition and template?

**Comment by johnd0e** [comment]
> What's the BBCode definition and template?

For example plain `[code]`, out-of-the-box.
phpbb 3.3.4
markdown 1.4.0

Post content:
```
[code]
[asdf]: zxcv
[/code]
```
Renders empty

**Comment by JoshyPHP** [comment]
Unfortunately, I don't think that specific interaction can be made to work without a complete redesign of the library. Plugins work independently and the Litedown plugin doesn't know that this link reference definition is inside of a BBCode code block. Normally it doesn't matter because markup is left as-is inside of code blocks but link reference definitions work differently.

The best workaround at the moment would be to use [Litedown code blocks](https://s9etextformatter.readthedocs.io/Plugins/Litedown/Syntax/#fenced-code-blocks), e.g.

~~~md
````
```
[asdf]: zxcv
```
````
~~~

**Comment by 3D-I** [comment]
Just for reference: https://www.phpbb.com/community/viewtopic.php?p=15739511#p15739511
It seems the OP is confusing markdown with MoonScript.

**Comment by johnd0e** [comment]
> Plugins work independently and the Litedown plugin doesn't know

That is exactly why I believe that #168 is worth resolving.

**Comment by JoshyPHP** [comment]
The issue is that BBCodes and Markdown (Litedown being a Markdown flavour) are fundamentally at odds with each other, because Markdown defines rules that are meant to apply to the whole document. BBCodes treat the text as a 1-dimensional stream of characters whereas Markdown requires the text processor to be mindful of columns and lines. BBCodes are mostly context-free whereas the meaning of some markup in Markdown depends on what precedes—or sometimes, follows—it.

For example, markup inside of a Markdown code block is sometimes interpreted and sometimes not, depending on the context. In the following example, some of the `>` inside of the blocks must be removed, others not. [See on Babelmark 2.](https://johnmacfarlane.net/babelmark2/?normalize=1&text=%60%60%60%0A%3E+%60%60%60%0A%60%60%60%0A%0A%3E+%60%60%60%0A%3E+%3E+%60%60%60%0A%3E+%60%60%60%0A%0A)
~~~md
```
> ```
```

> ```
> > ```
> ```
~~~

The short version is that the library is designed to treat the text as a stream of character and that plugins are independent of each other. Litedown cheats by keeping track of its own context/markup, but it cannot keep track of some special rules set up by other markup. That's why I say that it would require redesigning the library, which would take hundreds of hours.

On the other hand, using Litedown-style code blocks works fine and is available right now for free.

**Comment by johnd0e** [comment]
> On the other hand, using Litedown-style code blocks works fine and is available right now for free.

It is definitely enough for me personally, thanks.
But (for example) our board has a hundreds of active members, that are unaware of Litedown specifics.
So to make them less surprised I propose to make parsing rules slightly stricter #168.

**Comment by JoshyPHP** [comment]
I have investigated the feasibility of handling block-level Litedown markup inside of `code` BBCodes, and this cannot be done without a significant redesign. Since a workaround exists, I'll mark this as a "Won't fix" and will reevaluate if/when the relevant parts of the library are redesigned.

**Comment by johnd0e** [comment]
@JoshyPHP 
Ok, I understnd your position about code blocks.
But I have unwanted markdown parsing inside other bbcodes.
E.g. on our forum we have \[ascii\] bbcode, which has sort of similar meaning as code: raw monospaced text on dark background, like terminal:
```
[ascii]{TEXT}[/ascii]

<pre><div class="code_body" style="border: medium none; padding: 5px; color:silver; background:black none repeat scroll 0% 0%; font-size: 10pt; font-family: consolas;'courier new';terminal; line-height: 100%; letter-spacing: 0px; -moz-background-clip: -moz-initial; -moz-background-origin: -moz-initial; -moz-background-inline-policy: -moz-initial;">{TEXT}</div></pre>
```
[ascii]
4. a
5. b
[/ascii]
```
Is rendered as 
![image](https://user-images.githubusercontent.com/1838643/127518051-80524c29-da03-4e89-af0d-162f438c008b.png)

Is it possible to setup our `[ascii]` not to parse lists?
